### DISCUSSION TITLE
Ask help to connect roaster OzStar to Artisan

### DISCUSSION THREAD

**Discussion by ArtemUA81** [artisan-roaster-scope/artisan#1743] [Q&A]
I have turkish OzStar 2kg electric roaster, it is not in the list of brands wich Artisan support, but maybe it is possible to adjust it manual to start work with it?
![1](https://github.com/user-attachments/assets/a830e29f-ce40-4eb7-ae25-e58fd6925892)
![2](https://github.com/user-attachments/assets/abcb7b87-79cd-43dd-b047-c7c0645b5f1a)

**Comment by bigl00z3** [comment]
Dear @ArtemUA81 
not easy to give you a simple answer. I will start by a few questions:

- Do you have any documentation from the manufacturer on the communication protocol? 
- Do they mention if it is compatible with an external software (other than Artisan)?

Looking at your image, it seems that the PLC is a Schneider Modicon M221. I went on the [PLC webpage](https://www.se.com/fr/fr/product-range/62128-modicon-m221/#documents) and looked for the documentation. 

Based on what I saw in the documentation, it seems it uses the `Modbus TCP` to communicate which is supported by Artisan (see [here](https://artisan-scope.org/devices/modbus/) ).

However, for the PLC to be able to communicate with Artisan, you will need to configure everything yourself. 
For this you first need to know which Modbus registers are corresponding to which roaster parameter (for example a regisster number will correspond to the drum rpm, another one to the BT temperature...)  and also ensure that the PLC has been configured by the manufacturer to allow an external user to access/control those registers you are trying to reach.

If the manufacturer provides the documentation on which register you will save a lot of time. 

If you have zero documentation, you could try to find the Roaster IP, and then try to use the Artisan Port Scanner option to attempt to sniff the PLC but that can easily get very complicated if you don't know what you are doing.

I hope this can help you start

Nico

**Reply by roasterdave** [reply]
Thanks Nico for the helpful reply. Fortunately this machine is already covered.

**Reply by bigl00z3** [reply]
Well that's a good news.

**Reply by ArtemUA81** [reply]
Thnks, guys!
Will try all your advices!
Current setup with Twino dont give fast result((
I think to connect laptop I need to use switch or router to plug all together in one network. Next days will try to make it.
Thank you for your attention and information!
Later will write back and I hope with positive news))

**Comment by roasterdave** [comment]
There is a setup for this roaster available under Config>> Machine>> Twino Ozstar.

Easy to have missed because these roasters are marketed under both trade names, [Twino](https://www.twinocoffeeroasters.com) and [Ozstar](https://www.ozstarmakina.com).

More information about this setup can be found on the [Artisan Supported Machines](https://artisan-scope.org/machines/twino-ozstar/) page.
